tests/test_spatial_tools.py

```diff
@@ -178,6 +180,64 @@ def make_gdal_hillshade(filepath) -> np.ndarray:
     # Make sure that this doesn't create weird division warnings.
     xdem.spatial_tools.hillshade(dem.data, dem.res)
 
+class TestRobustFitting:
+
+    def test_robust_polynomial_fit(self):
+
+        np.random.seed(42)
+
+        # x vector
+        x = np.linspace(1,10,1000)
+        # exact polynomial
+        true_coefs = [-100, 5, 3, 2]
+        y = true_coefs[0] + true_coefs[1] * x + true_coefs[2] * x**2 + true_coefs[3] * x**3
+        # add some noise on top
+        y += np.random.normal(loc=0,scale=3,size=1000)
+        # and some outliers
+        y[50:75] = 0
+        y[900:925] = 1000
+
+        # test linear estimators
+        coefs, deg = xdem.spatial_tools.robust_polynomial_fit(x,y, estimator='Linear', linear_pkg='scipy', loss='soft_l1', f_scale=0.5)
+
+        # scipy solution should be quite robust to outliers/noise (with the soft_l1 method and f_scale parameter)
+        # however, it is subject to random processes inside the scipy function (couldn't find how to fix those...)
+        assert deg == 3 or deg == 4 # can find degree 3, or 4 with coefficient close to 0
```

Comment:
Here and below, the approximately same calculation is done with different estimators. Could this be simplified using `pytest.mark.parametrize`?

Reply:
Unfortunately the assert tests are all different, because it does not perform so well, so I think it will get harder to read with `pytest.mark.parametrize`.
But I should add a test case without outliers (where they all perform well) using `pytest.mark.parametrize` :)

Reply:
Added in 86d565b